src/components/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -286,18 +283,25 @@ const calendarDragAttributes = {
   },
 } as any
 
-const selectedCalendarRange = ref<TimeRange | Date | string>(props.modelValue)
-
-const hasCalendar = computed((): boolean => props.mode !== 'relative')
+// Booleans
+const hasCalendar = computed((): boolean => props.mode !== TimepickerMode.Relative)
+const isSingleDatepicker = computed((): boolean => ModeArrayCustom.includes(props.mode) && !props.range)
 const hasTimePeriods = computed((): boolean => props?.timePeriods?.length > 0)
 const showCalendar = computed((): boolean => state.tabName === 'custom' || !hasTimePeriods.value)
-const submitDisabled = computed((): boolean => {
-  // If either the calendar is in range selection mode, or relative time frames
-  // are present, check whether both `start` and `end` are set;
-  // Otherwise, it's a single date or time, so only check `start`
-  return props.range || hasTimePeriods.value
-    ? !state.selectedRange.start || !state.selectedRange.end
-    : !state.selectedRange.start
+const submitDisabled = ref<boolean>(true)
+
+// Dynamically choose the v-model since the Single date and Date range types are different
+const calendarSingleDate = ref<Date|null>(props.modelValue.start)
+const calendarRange = ref<TimeRange>(props.modelValue)
+const calendarVModel = isSingleDatepicker.value
+  ? calendarSingleDate as DatePickerModel
+  : calendarRange as DatePickerModel
+
+// `minute-increment` has been deprecated in favor of the time `rules` object
+// https://vcalendar.io/datepicker/time-rules.html
+const vCalendarRules = ref({
+  minutes: [0, 5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55],
+  seconds: [0],
```

Comment:
I noticed that `minute-increment` was no longer working, the sandbox timepickers were showing increments every 1 min.